src/main/java/org/svydovets/util/EntityReflectionUtils.java

```diff
@@ -87,4 +93,35 @@ public static Object newInstanceOf(Class<?> entityType) {
             );
         }
     }
+
+    public static boolean isEntityField(final Field field) {
+        boolean isEntityAnnotation = field.isAnnotationPresent(ManyToOne.class)
+                || field.isAnnotationPresent(OneToOne.class);
+        if (!field.isAnnotationPresent(JoinColumn.class)) {
+            throw new AnnotationMappingException(String.format("The entity field [%s] that is marked with the @OneToOne "
+                    + "or @ManyToOne annotation is missing the required @JoinColumn annotation", field.getName()));
+        }
```

Comment:
I think it's better to move this logic in the "resolveJoinColumnName()" method, because here we want just check if it's an entity. Besides this method will throw an exception if we pass a not entity field because value of "isEntityAnnotation" is not taken into account in your "if" statement